weave/tests/test_index_sql.py

```diff
@@ -0,0 +1,177 @@
+"""This script contains tests that test that pytest is deleting baskets
+correctly."""
+
+import os
+import sys
+
+# Try-Except required to make pyodbc an optional dependency.
+try:
+     # For the sake of explicitly showing that pyodbc is optional, import
+    # pyodbc here, even though it is not currently used in this file.
+    # Pylint ignore the next unused-import pylint warning.
+    # Also inline ruff ignore unused import (F401)
+    # pylint: disable-next=unused-import
+    import pyodbc # noqa: F401
+except ImportError:
+    _HAS_PYODBC = False
+else:
+    _HAS_PYODBC = True
+import pytest
+import s3fs
+from fsspec.implementations.local import LocalFileSystem
+
+from weave.index.index_sql import IndexSQL
+from weave.tests.pytest_resources import IndexForTest
+from weave.tests.pytest_resources import PantryForTest
+from weave.tests.pytest_resources import get_sample_basket_df
+
+s3fs = s3fs.S3FileSystem(
+    client_kwargs={"endpoint_url": os.environ["S3_ENDPOINT"]}
+)
+local_fs = LocalFileSystem()
+
+
+# Test with two different fsspec file systems (above).
+@pytest.fixture(
+    name="test_pantry",
+    params=[s3fs, local_fs],
+    ids=["S3FileSystem", "LocalFileSystem"],
+)
+def fixture_test_pantry(request, tmpdir):
+    """Sets up test pantry for the tests"""
+    file_system = request.param
+    test_pantry = PantryForTest(tmpdir, file_system)
+    yield test_pantry
+    test_pantry.cleanup_pantry()
+
+
+@pytest.fixture(
+    name="test_index",
+    params=[IndexSQL],
+    ids=["IndexSQL"],
+)
+def fixture_test_index(request):
+    """Sets up test index for the tests"""
+    index_constructor = request.param
+    test_index = IndexForTest(index_constructor, local_fs)
+    yield test_index
+    test_index.cleanup_index()
+
+
+# Skip tests if pyodbc is not installed.
+@pytest.mark.skipif(
+    "pyodbc" not in sys.modules or not _HAS_PYODBC
```

Comment:
aren't these first two kind of the same? if pyodbc is not in modules, then the _HAS_PYODBC flag should be false, right?

Reply:
Yes, but doing it this way also lets me avoid pylint unused complaint :joy: